### ISSUE TITLE
Does Not Apply CSS Custom Properties

### ISSUE THREAD

**Issue by nschomberg** [tmpvar/jsdom#1895]
### Basic info:

- Node **v8.1.2** 
- jsdom **v11.0.0** 

### Minimal reproduction case

```js
const { JSDOM } = require("jsdom");

const dom = new JSDOM(`<!DOCTYPE html><p>Hello world</p>`);

dom.window.document.querySelector(":root").style.setProperty('--var', 'red');
dom.window.document.querySelector(":root").style.setProperty('color', 'red');

console.log(dom.window.document.querySelector(":root").style);
// CSSStyleDeclaration {
//     '0': 'color',
//     _values: { color: 'red' },
//     _importants: { '--var': undefined, color: undefined },
//     _length: 1,
//     _onChange: [Function],
//     Var: 'red' }
```

### How does similar code behave in browsers?

Example: https://jsbin.com/xolomolexe/edit?js,console

In other browsers, CSS custom properties get applied like regular style properties.

**Comment by domenic** [comment]
Yeah, that's definitely not an implemented feature in our cssstyle/cssom packages. I know @Zirro is working on an ambitious rewrite of them that may lay the groundwork for things like this, but it's still a ways off. Sorry :-/.

**Comment by priley86** [comment]
@domenic is there currently any support for `:root` or `--var`? I'm curious about this now after a meeting today... ;)

**Comment by domenic** [comment]
Unfortunately no, sorry.

**Comment by tivac** [comment]
Checking in just to verify: is it still expected for code like this to silently fail in JSDOM 15? Not a complaint, just want to make sure I have correct expectations.

```js
const { JSDOM } = require("jsdom")

const dom = new JSDOM("<!DOCTYPE html><div>Hi</div>");

const div = dom.window.document.querySelector("div");

div.style.setProperty("--foo", "foo");

console.log(div.outerHTML);
```

**Comment by sheijne** [comment]
csstyle's CSSStyleDeclarations don't support custom properties. Seems the fix could be easy and should not need to wait for any "ambitious rewrite", correct me if I'm wrong.

https://github.com/jsdom/cssstyle/issues/89

**Comment by michaelfaith** [comment]
It seems this is still an issue.  Has anyone found a workaround for having `var()` calls resolve to the appropriate variable value?
